ttnmqtt/ttnmqtt.py

```
@@ -0,0 +1,185 @@
+                                        "GCM-SHA384:ECDHE-ECDSA-CHACHA20-"
+                                        "POLY1305:"
+                                        "ECDHE-RSA-CHACHA20-POLY1305:"
+                                        "ECDHE-ECDSA-AES128-GCM-SHA256:"
+                                        "ECDHE-RSA-AES128-GCM-SHA256:"
+                                        "ECDHE-ECDSA-AES256-SHA384:"
+                                        "ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-"
+                                        "AES128-SHA256:"
+                                        "ECDHE-RSA-AES128-SHA256")
+
+
+def _json_object_hook(d):
+    return namedtuple('MSG', d.keys())(*d.values())
+
+
+def json2obj(data):
+    return json.loads(data, object_hook=_json_object_hook)
+
+
+class DownlinkMessage:
+    def __init__(self, port, confirmed=False, schedule=None):
+        self.port = port
+        if confirmed:
+            self.confirmed = confirmed
+        if schedule:
+            self.schedule = schedule
+
+    def obj2json(self):
+        json_msg = json.dumps(self.__dict__)
+        return str(json_msg)
+
+
+class MyEvents(Events):
+    __events__ = (
+        "uplink_msg",
+        "downlink_msg",
+        "connect",
+        "close")
+
+
+class MQTTClient:
+
+    def __init__(self, appID, appAccessKey, **kwargs):
+        self.__client = mqtt.Client()
+        self.__appID = appID
+        self.__accessKey = appAccessKey
+        self.__events = MyEvents()
+        self.__mqttAddress = ""
+        self.__discoveryAddress = ""
+        for k, v in kwargs.items():
+            if k == "mqttAddress":
+                self.__mqttAddress = v
+            if k == "discoveryAddress":
+                self.__discoveryAddress = v
+        self.ErrorMsg = ""
+        try:
+            self.connect()
+            self.start()
+        except:
+            self.ErrorMsg = ("Connection failed: wrong appID,"
```

After thinking about it, I think we should let the exception be dropped, and document it - rather than catching it and having an `ErrorMsg` value:
+ The connection can fail for other reasons than wrong `appID`/`accessKey`/`mqttAddress`:
  + The machine may not have access to the network/The MQTT server could be down/Firewall restrictions could prevent connection
  + The client process doesn't have system capabilities to open a socket
  + The MQTT server uses MQTTS, but the client won't accept the TLS certificate
+ I don't think there are a lot of libraries that work with a system of `ErrorMsg` attribute. It could be counter-intuitive to many developers ; whereas exceptions is a part of the language.

Is there any specific reason why you'd like to keep `ErrorMsg` at all?